docs/index.md

```diff
@@ -2,5 +2,59 @@
 title: "Introduction"
 weight: 10
 ---
```

Comment:
hey @derberg heres the intro i had in mind. what do you think about it?

AsyncAPI Generator is a fast, feature-packed and ready to use Javascript based Npm library. We can use the
generator to generate whatever we want based on the AsyncAPI specification file. Generator has been built with 
extensibility in mind. You can generate anything out of it. Among other things, it can be used 
for generating code, documentation, diagrams, markdown files, microservices and applications. AsyncAPI offers a 
library of Templates which can be used to start generating code and docs immediately.

Reply:
I also think we should target the "what", "how" and "when" of the generator tool. 
The above text when finalised will decide What generator tool is?
@Florence-Njeri already has some awesome content regarding how generator works (generation process and related diags)

Do you think a section for How generator is useful makes sense in the intro document?🤔

Something like this:

Reply:
Generator helps organizations and developers speed up their project by automating the code and documentation generation process. Generator simplifies the generation process by providing us with a set of tools which help us produce our desired output in just a few steps! Generator promotes reusability. Once you have the AsyncAPI input file ready, you get access to multiple Templates. With minimum modification, you can generate code and documentation for different frameworks and languages.

Benefits of using Generator:

- Quick set up and easy to use on a regular basis
- Effortless generation of complex documents
- List of readily available AsyncAPI templates
- Support for render engines such as React and Nunjucks

cc: @Florence-Njeri @derberg